enhancements/20220120-select-another-disk-to-store-vm-data-during-installation.md

```diff
@@ -0,0 +1,190 @@
+  as the default disk on this Node, instead of `/var/lib/longhorn`.
+
+Another thing is we should prohibit v1.0.0 Nodes from joining a cluster with this change.
+See [Note](#version-incompatibility) for more detail.
+
+### Test plan
+
+1. Prepare a machine with two disks, let's say they will be `/dev/vda` and `/dev/sda`
+  during the installation.
+
+2. During the installation, select `/dev/vda` as the OS disk
+
+3. At the step "Remote config", provide the following config to use `/dev/sda`
+  as the default Longhorn disk:
+    ```
+    install:
+      data_disk: /dev/sda
+    ```
+
+4. After the installation finished, wait for the system to reboot and verify the following:
+    - `/dev/sda` is mounted to `/var/lib/harvester/defaultdisk` as the default disk.
+      Create a VM and verify Longhorn is storing data into `/var/lib/harvester/defaultdisk`.
+      You can create a huge file in the VM and see the space of `/dev/vda` is getting consumed.
+
+    - `/var/lib/longhorn` still exists, and Longhorn binaries are stored here.
+        ```
+        $ ls -l /var/lib/longhorn/engine-binaries
+        total 4
+        drwxr-xr-x 2 root root 4096 Jan 19 08:14 longhornio-longhorn-engine-v1.2.3
+        ```
+
+5. Prepare another machine to join the cluster. Again you need two disks on this one.
+
+6. Provide the same config as step 3 to use the other disk as the default Longhorn disk.
+
+7. Perform the same verification as step 4 on this join Node.
+
+### Upgrade strategy
+
+No user intervention is required during the upgrade.
+
+## Note
+
+### Supporting select multiple disks to store VM data
+Longhorn supports creating multiple default disks with Node-wise settings:
+
+- Change the Longhorn setting "createDefaultDiskLabeledNodes" to `true`.
+
+- Add a label to the Node: `node.longhorn.io/create-default-disk="config"`,
+  indicating default disks on this Node should be created with a Node-wise config.
+
+- Add an annotation to the Node: `node.longhorn.io/default-disks-config=<disks config in JSON>`
+  to specify the Node-wise config.
+
+Then Longhorn would follow the `<disks config in JSON>` to create default disks
+[REF](https://longhorn.io/docs/1.2.3/advanced-resources/default-disk-and-node-config/#customizing-default-disks-for-new-nodes).
+
+What matters here is we can't add annotations to the Node during installation.
+It's possible to add labels, but not annotations.
+
+Luckily, the labels and annotations could be added to the Node after it has joined the cluster.
+Longhorn would follow the annotation to create default disks once it appears,
+**as long as default disks have never been created on the Node.**
+Therefore, we should be able to rely on NDM or other controllers to
+help add the labels and annotations after it joined the cluster, but it requires extra effort.
+
+### Version incompatibility
+We should prohibit users from adding v1.0.0 Nodes into the cluster with this enhancement.
```

Comment:
Very good catch. We should add `require/release-note` to the original GitHub issue to call it out in the release note.